src/pke/lib/schemerns/rns-leveledshe.cpp

```diff
@@ -422,25 +446,25 @@ Ciphertext<DCRTPoly> LeveledSHERNS::EvalMultMutable(Ciphertext<DCRTPoly>& cipher
         return EvalMultCore(ciphertext, pt);
     }
 
+    auto ctmorphed = MorphPlaintext(plaintext, ciphertext);
+
     if (cryptoParams->GetScalingTechnique() == FIXEDMANUAL) {
-        Ciphertext<DCRTPoly> c = ciphertext->Clone();
-        DCRTPoly pt            = AdjustLevelsInPlace(c, plaintext);
-        auto result            = EvalMultCore(c, pt);
-        result->SetScalingFactor(c->GetScalingFactor() * plaintext->GetScalingFactor());
-        result->SetDepth(c->GetDepth() + plaintext->GetDepth());
+        Ciphertext<DCRTPoly> ct = ciphertext->Clone();
```

Comment:
The function should be re-written as logic is the same for  FIXEDMANUAL and for the block after. we just need to have an auto reference to the correct ciphertext.